detex.l

```diff
@@ -665,6 +691,65 @@ yyless(int n)
 }
 #endif
 
+/******
+** LineBreak -- choses the proper way to break a line. If '-1' option is
+**      enabled we also want to output some source location information.
+******/
+
+void
+LineBreak()
+{
+	if (fWord) return;
+	if (fSrcLoc && fIsColumn0) {
+		printf("%s:%4d: ", fFileNames[csb], fFileLines[csb]);
+		fIsColumn0 = 0;
+	}
+	putchar('\n');
+	fFileLines[csb]++; fIsColumn0=1;
+}
+
+/******
+** Echo -- If we are at column 0 and have specified '-1'; output 
+**      the source location information.
+******/
+
+void
+Echo()
+{
+	if (fSrcLoc && fIsColumn0) {
```

Comment:
These four lines are duplicated in `LineBreak`. Can you extract them to a function?